src/ngMessages/messages.js

```diff
@@ -102,6 +125,60 @@
  * email addresses, date fields, autocomplete inputs, etc...), specialized error messages can be applied
  * while more generic messages can be used to handle other, more general input errors.
  *
+ * ## Dynamic Messaging
+ * In AngularJS 1.4 ngMessages supports using expressions to dynamically change key values. Using arrays
+ * and repeaters to list messages is also supported. This means that the code below will be able to fully
+ * adapt itself and display the appropriate message when any of the expression data changes:
+ *
+ * ```html
+ * <form name="myForm">
+ *   <input type="email"
+ *          name="myEmail"
+ *          ng-model="email"
+ *          minlength="5"
+ *          required />
+ *
+ *   <div ng-messages="myForm.myEmail.$error">
+ *     <div ng-message="required">You did not enter your email address</div>
+ *     <div ng-repeat="errorMessage in errorMessages">
+ *       <!-- use ng-message-exp for a message that is an expression -->
+ *       <div ng-message-exp="errorMessage.type">{{ errorMessage.text }}</div>
+ *     </div>
+ *   </div>
+ * </form>
+ * ```
+ *
+ * The `errorMessage.type` variable can be a string value or it can be an array that can have
+ * multiple error states be associated with a single error message:
```

Comment:
... or it can be an array so that multiple error states can be associated with a single error message: